conformance/src/txn_execute.zig

```diff
@@ -860,25 +862,32 @@ fn serializeOutput(
                 .fees_only => false,
             };
 
-            // TODO: just hardcoding InstructionError for now, can something else happen?
-            // our current design assumes that the transaction execution can only return
-            // InstructionError, which seems wrong. Not to mention that we don't have a way
-            // to access *which* instruction in the transaction is the one that errored.
-            const status: u32, const instr_err: u32 = switch (txn) {
-                .executed => |executed| .{
-                    if (executed.executed_transaction.err != null) 9 else 0,
-                    if (executed.executed_transaction.err) |instr_err|
-                        @intFromEnum(instr_err) + 1
+            const converted = blk: {
+                // https://github.com/firedancer-io/solfuzz-agave/blob/agave-v2.2.13/src/txn_fuzzer.rs#L208-L210
+                const txn_error: TransactionError = switch (txn) {
+                    .executed => |executed| if (executed.executed_transaction.err) |instr_err_enum|
+                        .{ .InstructionError = .{
+                            executed.executed_transaction.err_index,
+                            instr_err_enum,
+                        } }
                     else
-                        0,
-                },
-                .fees_only => |fees_only| fees_only: {
-                    const err_codes = utils.convertTransactionError(fees_only.err);
-                    break :fees_only .{
-                        err_codes.err,
-                        err_codes.instruction_error,
-                    };
-                },
+                        break :blk std.mem.zeroes(utils.Converted),
+                    .fees_only => |fees_only| fees_only.err,
+                };
+
+                // Set custom err to 0 if the failing instruction is a precompile
+                var converted = utils.convertTransactionError(txn_error);
+                if (converted.instruction_index < sanitized.instruction_infos.len) {
+                    const ixn_info = &sanitized.instruction_infos[converted.instruction_index];
+                    for (sig.runtime.program.precompiles.PRECOMPILES) |precompile| {
+                        if (precompile.program_id.equals(&ixn_info.program_meta.pubkey)) {
+                            converted.custom_error = 0;
+                            break;
+                        }
+                    }
+                }
+
+                break :blk converted;
```

Comment:
just a note, these changes are going to conflict with #840 which makes some similar fixes, except it appears to be missing this precompile check; may be helpful to coordinate on this cc @yewman.